scipy/sparse/tests/test_construct.py

```diff
@@ -373,6 +375,20 @@ def test_bmat(self):
                             r'Got blocks\[0,1\]\.shape\[0\] == 1, expected 2',
                             construct.bmat, [[A,C]])
 
+    @pytest.mark.slow
+    def test_concatenate_int32_overflow():
+        """ test for indptr overflow when concatenating matrices """
+        check_free_memory(30000)
+        
+        n = 33000
+        A = csr_matrix(np.ones((n, n)), dtype=bool))
+        B = A.copy()
+        C = construct._compressed_sparse_stack((A,B), 0)
```

Comment:
I was using `_compressed_sparse_stack` on purpose as the bug is not in `vstack` (and may also appear in `hstack`)